### PR TITLE
Display sections in UserSettings separately

### PR DESCRIPTION
**Pull request by rszwajko**
Motivation: the list of section has recently grown too long. User is forced to scroll page to see all options. Future enhancements will add even more content so it's time to split the sections. 

Design: follow the original design [1]  updated to take into consideration newest PF recomendations [2] (i.e. arrow should be present if there is nested navigation items).

[1]  https://marvelapp.com/prototype/94h0g56/screen/59286966
[2]  https://www.patternfly.org/v4/components/navigation

**Screenshots**:
section_general
section_refresh
section_notifications

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by rszwajko** [comment]
ci add to whitelist

**Comment by rszwajko** [comment]
ci test please

**Comment by sjd78** [comment]
ci add to whitelist

**Comment by sjd78** [comment]
ci test

File: src/components/UserSettings/style.css [hunk 1]
```diff
@@ -1,3 +1,12 @@
 .half-width {
     width: 50%;
+}
+
+:global(.pf-c-nav__list.card-pf) {
```
Comment (sjd78) [inline src/components/UserSettings/style.css:5]:
```suggestion
.nav-list-container {
```

And then in `GlobalSettings` do 
```jsx
  <NavList className={style['nav-list-container']}>
```

File: src/components/UserSettings/style.css [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+    padding: 0;
+    margin-top: 20px;
+}
+
+.border {
+    border-bottom:  1px #0000001f
 }
```
Comment (sjd78) [inline src/components/UserSettings/style.css:12]: Nit-pick: fix EOL/EOF

**Review by sjd78 — CHANGES_REQUESTED** [review]
Code looks ok, but suggesting a css change

File: src/components/UserSettings/style.css (hunk 1 above)
Comment (rszwajko) [reply src/components/UserSettings/style.css]: We need to override `card-pf` but it's definitely better to narrow the scope as much as possible. What do you think about adding a marker class just for this purpose? Take a look at newest version.

**Comment by rszwajko** [comment]
Changes:
1. added marker class to prevent css side effects in other places
2. renamed prop `key` (React restricted) to `name`

File: src/components/UserSettings/style.css (hunk 1 above)
Comment (sjd78) [reply src/components/UserSettings/style.css]: Yeah the extra marker class looks good.

File: src/components/UserSettings/style.css (hunk 2 above)
Comment (sjd78) [reply src/components/UserSettings/style.css:13]: So no extra newline there?

**Review by sjd78 — COMMENTED** [review]
Looking good except for that last EOF/EOL nit-pick.

**Comment by sjd78** [comment]
ci test

File: src/components/UserSettings/style.css (hunk 2 above)
Comment (rszwajko) [reply src/components/UserSettings/style.css]: hmmm ...it got lost again:)

**Comment by rszwajko** [comment]
ci test

**Review by sjd78 — APPROVED** [review]
LGTM

**Comment by rszwajko** [comment]
force push to resolve merge conflict with #1424

File: yarn.lock [hunk 3]
```diff
@@ -8043,7 +7874,12 @@ lodash.words@^3.0.0:
   dependencies:
     lodash._root "^3.0.0"
 
-lodash@4.17.21, lodash@^4.13.1, lodash@^4.14.2, lodash@^4.17.10, lodash@^4.17.11, lodash@^4.17.12, lodash@^4.17.13, lodash@^4.17.14, lodash@^4.17.15, lodash@^4.17.2, lodash@^4.17.4, lodash@^4.17.5, lodash@^4.2.0, lodash@^4.3.0:
+lodash@4.17.15:
+  version "4.17.15"
```
Comment (sgratch) [inline yarn.lock:7878]: This is not good due to a CVE required us to update lodash to >= 4.17.21.
Check https://github.com/oVirt/ovirt-web-ui/pull/1413
Please revert back this section if possible for using 4.17.21 instead.

**Review by sgratch — COMMENTED** [review]
Looks great to me except the lodash version dependency.

Two UX improvements/issues to consider:
- Can we set the order of fields within the user settings confirmation dialog to fit fields/tabs order?
image

- Do you think it is clear for the user to understand that all actions (save/Cancel/Reset settings) are applied to all tabs in once and not only the current one? Or should we somehow reflect that within the confirmation dialogs?

File: yarn.lock (hunk 3 above)
Comment (rszwajko) [reply yarn.lock:7878]: Fixed in #1437

**Comment by rszwajko** [comment]
> Can we set the order of fields within the user settings confirmation dialog to fit fields/tabs order?
>

Yes.

> 
>     * Do you think it is clear for the user to understand that all actions (save/Cancel/Reset settings) are applied to all tabs in once and not only the current one? Or should we somehow reflect that within the confirmation dialogs?

I think we could also group it by section.

**Review by sgratch — COMMENTED** [review]
@rszwajko 
On settings dialog - the font size, line height and font type of field labels and nav-items are now changed and are not compatible any more with other web-ui dialogs (font size is changed from 12px to 16px).

Before:
image

After:
image

**Comment by sgratch** [comment]
> > Can we set the order of fields within the user settings confirmation dialog to fit fields/tabs order?
> 
> Yes.

+1

> 
> > Do you think it is clear for the user to understand that all actions (save/Cancel/Reset settings) are applied to all tabs in once and not only the current one? Or should we somehow reflect that within the confirmation dialogs?
> 
> I think we could also group it by section.

Not sure it worth the effort, maybe just mentioning it within a tooltip/confirmation dialog is enough

Let's open a separate issues for both

**Comment by rszwajko** [comment]
@sgratch 
> On settings dialog - the font size, line height and font type of field labels and nav-items are now changed and are not compatible any more with other web-ui dialogs (font size is changed from 12px to 16px).

I cannot reproduce it on my env. Please provide more details about your env.

**Comment by sgratch** [comment]
> @sgratch
> 
> > On settings dialog - the font size, line height and font type of field labels and nav-items are now changed and are not compatible any more with other web-ui dialogs (font size is changed from 12px to 16px).
> 
> I cannot reproduce it on my env. Please provide more details about your env.

I can't reproduce it now as well :) 
Not sure what caused that problem to me, I did try to clean my cache when it was reproduced. Sorry for the confusion

Since not reproduced then this PR LGTM.
a followup issues were opened for https://github.com/oVirt/ovirt-web-ui/pull/1427#issuecomment-851608502:
https://github.com/oVirt/ovirt-web-ui/issues/1439
https://github.com/oVirt/ovirt-web-ui/issues/1440

**Pull request merged**